src/utils/googleTagManager.ts

```diff
@@ -155,14 +165,57 @@ export const trackEvent = ({
   action: string
   label?: EventLabel
 }): void => {
-  const dataLayer = {
+  const dataLayer: EventDataLayer = {
     event,
     chainId: _getChainId(),
     eventCategory: category,
     eventAction: action,
     eventLabel: tryParse(label),
   }
 
+  track(dataLayer)
+}
+
+type SafeAppEventDataLayer = {
+  event: GTM_EVENT
+  chainId: string
+  safeAppName: string
+  safeAppMethod: string
+  safeAppParams?: string
+  safeAppEthMethod?: string
+  safeAppSDKVersion?: string
+}
+
+export const getSafeAppName = (safeApp?: SafeApp): string => {
+  const parsedSafeApp = JSON.parse(safeApp?.id || 'Unknown Safe App')
```

Comment:
Please extract `Unknown Safe App`. The relevant unknown string is also assigned within the `AppFrame`. I would use the same value there.

Reply:
where in AppFrame?

Reply:
Sorry, it's not in `AppFrame` but `src/routes/safe/components/Apps/utils.ts`, `EMPTY_SAFE_APP`.